pkg/ccl/serverccl/diagnosticsccl/reporter_test.go

```diff
@@ -366,6 +374,29 @@ func TestTelemetry_SuccessfulTelemetryPing(t *testing.T) {
 
 }
 
+// This test will block on `stopper.Stop` if the diagnostics reporter
+// doesn't honor stopper quiescence when making its HTTP request.
+func TestTelemetryQuiesce(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	defer setTelemetryHttpTimeout(100 * time.Second)()
+	rt := startReporterTest(t, base.TestIsSpecificToStorageLayerAndNeedsASystemTenant)
+	defer rt.Close()
+
+	ctx := context.Background()
+	setupCluster(t, rt.serverDB)
+
+	defer rt.diagServer.SetWaitSeconds(200)()
+	dr := rt.server.DiagnosticsReporter().(*diagnostics.Reporter)
+	stopper := rt.server.Stopper()
+
+	dr.PeriodicallyReportDiagnostics(ctx, stopper)
+	stopper.Stop(ctx)
+	<-stopper.IsStopped()
```

Comment:
Not sure if bors is going to merge this, but might be worth checking some upper bound:

```go
	dr.PeriodicallyReportDiagnostics(ctx, stopper)
	stopper.Stop(ctx)
	select {
	   case <-time.After(time.Second):
	     t.Fatal("expected stopper to signal")
	   case <-stopper.IsStopped9):
	     t.Log("stopped in time")
   }
```

Reply:
That should get captured by the standard test timeouts which trigger in case of slow quiescence or arbitrary blocking. I'll just bump the wait times to 5 minutes to make sure that happens.

Reply:
makes sense, that sounds reasonable to me